drivers/watchdog/wdt_max32.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2023 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_max32_watchdog
+
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/clock_control/adi_max32_clock_control.h>
+#include <soc.h>
+#include <errno.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_max32, CONFIG_WDT_LOG_LEVEL);
+
+#include <wrap_max32_wdt.h>
+
+struct max32_wdt_config {
+	mxc_wdt_regs_t *regs;
+	const struct device *clock;
+	struct max32_perclk perclk;
+	void (*irq_func)(void);
+};
+
+struct max32_wdt_data {
+	struct wdt_window timeout;
+	wdt_callback_t callback;
+};
+
+static int wdt_max32_calculate_timeout(uint32_t timeout, uint32_t clock_src)
+{
+	int i;
+	uint32_t clk_frequency = 0;
+	uint32_t number_of_tick = 0;
+
+	switch (clock_src) {
+	case ADI_MAX32_PRPH_CLK_SRC_PCLK:
+		clk_frequency = PeripheralClock;
+		break;
+
+	case ADI_MAX32_PRPH_CLK_SRC_EXTCLK:
+		clk_frequency = ADI_MAX32_CLK_EXTCLK_FREQ;
+		break;
+
+	case ADI_MAX32_PRPH_CLK_SRC_IBRO:
+		clk_frequency = ADI_MAX32_CLK_IBRO_FREQ;
+		break;
+
+	case ADI_MAX32_PRPH_CLK_SRC_ERFO:
+		clk_frequency = ADI_MAX32_CLK_ERFO_FREQ;
+		break;
+
+	case ADI_MAX32_PRPH_CLK_SRC_ERTCO:
+		clk_frequency = ADI_MAX32_CLK_ERTCO_FREQ;
+		break;
+
+	case ADI_MAX32_PRPH_CLK_SRC_INRO:
+		clk_frequency = ADI_MAX32_CLK_INRO_FREQ;
+		break;
+
+	default:
+		LOG_ERR("Unsupported clock source.");
+		return -ENOTSUP;
```

Comment:
As clock_src is an enum, this can't happen.  So you could remove this, and the callers don't need to check for -ENOTSUP.
Is PeripheralClock a #define?  If so you could use an array rather than a switch/case.

Reply:
Updated that part with using a defined macro. Checked equal to 0 case because used macro can return 0.